### PR TITLE
feat: logout with multiple accounts epic

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

logout with multiple accounts

### Solutions

this PR is a collection of multiple smaller PRs:
1. #945 (this one is old since AuthSession is deleted)
2. #953 
3. #976 

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-982/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4419621)(4419621)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4419621)(4419621)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/feature/AccountSwitchUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+import javax.inject.Singleton
+
+@Suppress("LongParameterList")
+@Singleton
+class AccountSwitchUseCase @Inject constructor(
+    private val updateCurrentSession: UpdateCurrentSessionUseCase,
+    private val navigationManager: NavigationManager,
+    private val getSessions: GetSessionsUseCase,
+    private val getCurrentSession: CurrentSessionUseCase,
+    private val deleteSessionUseCase: DeleteSessionUseCase,
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/feature/AccountSwitchUseCase.kt]:
```suggestion
    private val deleteSession: DeleteSessionUseCase,
```

**Review by yamilmedina — APPROVED** [review]
Small thing! Great cherry on top 🍒 !

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-982/5/) **failed**.

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-982/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4420001)(4420001)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4420001)(4420001)-dev-release.apk)

**Pull request merged**